### PR TITLE
Add support for unhealthy healthcheck result with custom message and …

### LINKED ISSUE #887: HealthCheck should allow a custom message when built from a throwable
Make it possible to specify a custom error message when constructing a new unhealthy Result:
e.g. Result.unhealthy("my custom message", theThrowable);

### PR DESCRIPTION
**Pull request by phillbarber**
…exception. Closes #887

[Review on Reviewable](https://reviewable.io/reviews/ratpack/ratpack/888)

### CODE REVIEW

File: ratpack-core/src/main/java/ratpack/health/HealthCheck.java [hunk 1]
```diff
@@ -154,7 +154,20 @@ public static Result unhealthy(String message, Object... args) {
...
+    }
+
+    /**
+     * Creates an unhealthy result, with the given exception and message.
+     * <p>
+     * The supplied message will be used as the message of the result.
+     *
+     * @param message a message to accompany the result
+     * @param error an exception thrown during health check
+     * @return an unhealthy result, with the given error and message
```
Comment (alkemist) [inline ratpack-core/src/main/java/ratpack/health/HealthCheck.java]: Can you please add `@since 1.2` here.

**Comment by alkemist** [comment]
Many thanks @phillbarber 

Can you please add that `@since` mentioned in the line comment and add yourself to https://github.com/ratpack/ratpack/blob/b327d050ff906ec8d13118773c317dd4c3a8d9ce/ratpack-manual/src/content/chapters/99-about-the-project.md#L82-L82

**Comment by phillbarber** [comment]
Thanks @alkemist Have also added @sebbonnet to the contributers as we were actually pairing on this.

**Comment by alkemist** [comment]
Fantastic. Thanks again.

**Pull request merged**